tests/fixtures/vector_io.py

```diff
@@ -240,3 +256,87 @@ def get_etcd_deployment_template() -> Dict[str, Any]:
             ],
         },
     }
+
+
+@pytest.fixture(scope="class")
+def pgvector_deployment(
+    unprivileged_client: DynamicClient,
+    unprivileged_model_namespace: Namespace,
+) -> Generator[Deployment, Any, Any]:
+    """Deploy a PGVector instance for vector I/O provider testing."""
+    with Deployment(
+        client=unprivileged_client,
+        namespace=unprivileged_model_namespace.name,
+        name="vector-io-pgvector-deployment",
+        min_ready_seconds=5,
+        replicas=1,
+        selector={"matchLabels": {"app": "pgvector"}},
+        strategy={"type": "Recreate"},
+        template=get_pgvector_deployment_template(),
+        teardown=True,
+    ) as deployment:
+        deployment.wait_for_replicas(deployed=True, timeout=240)
+        yield deployment
+
+
+@pytest.fixture(scope="class")
+def pgvector_service(
+    unprivileged_client: DynamicClient,
+    unprivileged_model_namespace: Namespace,
+    pgvector_deployment: Deployment,
+) -> Generator[Service, Any, Any]:
+    """Create a service for the PGVector deployment."""
+    with Service(
+        client=unprivileged_client,
+        namespace=unprivileged_model_namespace.name,
+        name="vector-io-pgvector-service",
+        ports=[
+            {
+                "name": "postgres",
+                "port": 5432,
+                "targetPort": 5432,
+            },
+        ],
+        selector={"app": "pgvector"},
+        wait_for_resource=True,
+    ) as service:
+        yield service
+
+
+def get_pgvector_deployment_template() -> Dict[str, Any]:
+    """Return a Kubernetes deployment for PGVector"""
+    return {
+        "metadata": {"labels": {"app": "pgvector"}},
+        "spec": {
+            "containers": [
+                {
+                    "name": "pgvector",
+                    "image": PGVECTOR_IMAGE,
+                    "ports": [{"containerPort": 5432}],
+                    "env": [
+                        {"name": "POSTGRES_DB", "value": "pgvector"},
+                        {"name": "POSTGRES_USER", "value": "vector_user"},
+                        {"name": "POSTGRES_PASSWORD", "value": "yourpassword"},
+                        {"name": "PGDATA", "value": "/var/lib/postgresql/data/pgdata"},
+                    ],
+                    "lifecycle": {
+                        "postStart": {
+                            "exec": {
+                                "command": [
+                                    "/bin/sh",
+                                    "-c",
+                                    (
+                                        "sleep 5\n"
+                                        "PGPASSWORD=yourpassword psql -h localhost -U vector_user "
```

Comment:
I think that user is created automatically without any command and pgvector extension is activated by default in pgvector images.

Reply:
It seems that even on a pgvector image this step is still necessary, I got this error when I removed it (Llama Stack side):
``` bash
RuntimeError: Vector extension is not installed.
Traceback (most recent call last):
File "/opt/app-root/lib64/python3.12/site-packages/llama_stack/providers/remote/vector_io/pgvector/pgvector.py", line 357, in initialize
raise RuntimeError("Vector extension is not installed.")
RuntimeError: Vector extension is not installed.
```